collections/sequence.go

```diff
@@ -0,0 +1,51 @@
+package collections
+
+import (
+	"context"
+	"errors"
+)
+
+// DefaultSequenceStart defines the default starting number of a sequence.
+const DefaultSequenceStart uint64 = 1
+
+// Sequence builds on top of an Item, and represents
+// an ever-increasing number.
+type Sequence Item[uint64]
```

Comment:
Naive q: what is the usecase for this type

Reply:
It's a common type, EG: auth sequence, everything that uses an ever increasing unique ID.